backends/bmv2/psa_switch/options.h

```diff
@@ -0,0 +1,43 @@
+/*
+Copyright 2013-present Barefoot Networks, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#ifndef BACKENDS_BMV2_PSA_SWITCH_OPTIONS_H_
+#define BACKENDS_BMV2_PSA_SWITCH_OPTIONS_H_
+
+#include "backends/bmv2/common/options.h"
+#include "backends/bmv2/psa_switch/midend.h"
+
+namespace BMV2 {
+
+class PsaSwitchOptions : public BMV2Options {
+ public:
+    PsaSwitchOptions() {
+        registerOption("--listMidendPasses", nullptr,
+                [this](const char*) {
+                    listMidendPasses = true;
+                    loadIRFromJson = false;
+                    PsaSwitchMidEnd midEnd(*this, outStream);
```

Comment:
this is a bit strange; I would have expected that you will pass the flag to the existing midend.
But I guess that this works.

Reply:
I assumed that p4c should be able to run with single option '--listMidendPasses', similar to '--help' option. 
If so, there is no input file to compile and the best way to get to the midend passes names 
instantly is to call midend constructor.